src/components/GoodsList.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from 'react';
+import {
+  goodsFromServer,
+  renderList,
+  filterGoodsByLength,
+} from './Helpers';
+
+const GoodsList: React.FC = () => {
+  const [isClicked, setIsClicked] = useState(false);
+  const [goodsList, setGoodsList] = useState([...goodsFromServer]);
+  const [defaultLength, setDefaultLength] = useState(true);
+
+  const sortAlphabetically = (goods: string[]) => {
+    setGoodsList([...goods].sort((goodItem1: string, goodItem2: string) => (
+      goodItem1.localeCompare(goodItem2)
+    )));
+
+    return renderList(goodsList);
+  };
+
+  const reset = () => {
+    setGoodsList([...goodsFromServer]);
+    setDefaultLength(true);
+
+    return renderList(goodsList);
+  };
+
+  const reverse = (goods: string[]) => {
+    setGoodsList(
+      [...goods].reverse(),
+    );
+
+    return renderList(goodsList);
+  };
+
+  const sortByLength = (goods: string[]) => {
+    const sortedGoods = [...goods].sort((goodItem1, goodItem2) => (
+      goodItem1.length - goodItem2.length
+    ));
+
+    setGoodsList(
+      sortedGoods,
+    );
+
+    return renderList(sortedGoods);
+  };
+
+  const startButton = isClicked
+    ? null
+    : (
+      <button
+        className="button is-primary"
+        type="button"
+        onClick={() => setIsClicked(true)}
+      >
+        Start
+      </button>
+    );
+
+  return (
+    <>
+      <h1 className="title is-1">Goods</h1>
+      {startButton}
+      <article className="content is-medium">
+        <ul>
+          {
+            (isClicked)
+            && renderList(goodsList)
+          }
+        </ul>
+      </article>
+
+      <hr />
+      <article className="buttons">
+        <button
+          disabled={!isClicked}
+          className="button is-dark"
+          type="button"
+          onClick={() => reverse(goodsList)}
+        >
+          Reverse
+        </button>
+
+        <button
+          disabled={!isClicked}
+          className="button is-dark"
+          type="button"
+          onClick={() => sortAlphabetically(goodsList)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          disabled={!isClicked}
+          className="button is-dark"
+          type="button"
+          onClick={() => reset()}
+        >
+          Reset
+        </button>
+
+        <button
+          disabled={!isClicked}
+          className="button is-dark"
+          type="button"
+          onClick={() => sortByLength(goodsList)}
+        >
+          Sort by length
+        </button>
+
+        <div className="select is-rounded is-primary">
+          <select
+            disabled={!isClicked}
+            name="filterOptions"
+            id="filterOptions"
+            onChange={
+              (event) => {
+                setGoodsList(filterGoodsByLength(+event.target.value));
+                setDefaultLength(false);
+              }
+            }
+          >
+            <option value="1" selected={defaultLength}>1</option>
```

Comment:
essentially `selected` could sit here standalone
```suggestion
            <option value="1" selected>1</option>
```